fix for not detecting the user language

cvgs opened the pull request:
Another side effect of the RecordName ambiguity: RecordName and HomeDir do not necessary match. By explicitly getting the NFSHomeDirectory attribute, we can get the real path to the user's home directory and then safely read the current AppleLanguages setting.

Otherwise we could see a PlistBuddy error message in the log and would default to english even the OS language was set to german.

MichalMMac commented on erase-install.sh:
```diff
@@ -74,7 +74,9 @@ catalogurl="https://swscan.apple.com/content/catalogs/others/index-12-10.16-10.1
 current_user=$(/usr/sbin/scutil <<< "show State:/Users/ConsoleUser" | /usr/bin/awk -F': ' '/[[:space:]]+Name[[:space:]]:/ { if ( $2 != "loginwindow" ) { print $2 }}')
-language=$(/usr/libexec/PlistBuddy -c 'print AppleLanguages:0' "/Users/${current_user}/Library/Preferences/.GlobalPreferences.plist")
+# Get proper home directory. Output of scutil might not reflect the canonical RecordName or the HomeDirectory at all, which might prevent us from detecting the language
+current_user_homedir=$(/usr/bin/dscl /Search -read /Users/${current_user} NFSHomeDirectory | /usr/bin/awk -F': ' '{print $2}')
```
Using `awk` on `dscl` output is not safe.  Sometimes you can get the value on different line than the key. Safe way to do it is to use `-plist` option for `dscl` and get the value from the plist.

cvgs commented:
This is true.

I could rewrite it as follows:
`current_user_homedir=$(/usr/bin/dscl -plist -q /Search -read /Users/${current_user} NFSHomeDirectory | /usr/bin/xmllint --nonet --xpath '//key[.="dsAttrTypeStandard:NFSHomeDirectory"]/following-sibling::*[1]/string/text()' -)`

But my previous pull request also uses dscl and awk, because i have not yet found a way to parse the multi-line output for RecordName with xmllint, as it does not output field separators:
 
`/usr/bin/dscl -plist -q /Search -read Users/$user RecordName dsAttrTypeStandard:RecordName`

If 'jq' was preinstalled on macOS, this would be easy, but that's not the way it is.

Dumping the output to a temporary file and doing multiple "defaults read" also sounds like an ugly idea. Perhaps i could use osascript/Javascript and its JSON power to do plist evaluation? But dscl outputs XML not JSON, so we still would need to convert that.

Any pointers? Guess i need to think that through over the holidays…

MichalMMac commented:
What about:

```
/usr/libexec/PlistBuddy -c 'Print :dsAttrTypeStandard\:NFSHomeDirectory:0' /dev/stdin <<< "$(/usr/bin/dscl -plist /Search -read /Users/michal NFSHomeDirectory)"
```

via https://scriptingosx.com/2018/07/parsing-dscl-output-in-scripts/

ryangball commented:
If you are worried about it being on the next line you could easily use tail to account for both varieties and skip a bunch of code:
`current_user_homedir=$(/usr/bin/dscl . -read "/Users/${current_user}" NFSHomeDirectory | /usr/bin/tail -n 1 | /usr/bin/awk '{print $NF}')`

cvgs commented:
I've incorporated the suggestion by @MichalMMac and changed the variable naming to better fit into the existing style and added those commits. The modifications work in my tests.

grahampugh commented:
Thanks
